docs/general/adapters.md

```diff
@@ -36,6 +36,20 @@ The `Attributes` adapter does not include a root key. It is just the serialized
 
 Use either the `JSON` or `JSON API` adapters if you want the response document to have a root key.
 
+To automatically switch between adapters based on the request, set the adapter to `:mime`
+
+```ruby
+  # In configuration
+  ActiveModelSerializers.config.adapter = :mime
+  
+  # In controller action
+  respond_to do |format|
+    format.json { render json: Model.first }     # Uses the JSON adapter
+    format.json_api { render json: Model.first } # Users the JSON:API adapter
```

Comment:
👍 

should be `jsonapi` not `json_api`, and `render jsonapi: model` rather than `render json: model`.

But really, this is just using the content type negotiation that respond_to already does. This would work without any changes to the PR, assuming mime -> adapter is 1:1